### ISSUE TITLE
code for deepnet & reproduction

### ISSUE THREAD

**Issue by KennyShang** [microsoft/unilm#663]
**Need code of deepnet for reproduction**
Failed to reproduce the deepnet paper with the TL:DR section

**Modifications for post-ln tranformers**
1. calculate the alpha and beta for encoder and decoder;
1. Up scale the residual x with beta;
2. Reinit the weights in the q k v output and ffn according to paper;

**Expected behavior**
After 30k training steps, the Bleu score decreases to 0 from 24-25 in a model with a config of  35L-35L;

**Comment by maydaygmail** [comment]
Meet the similar problem. bleu score decrease to 0 after about 10K training step.

**Comment by shumingma** [comment]
Hi @KennyShang @maydaygmail Could you provide more details about your implementation (e.g. **alpha, beta** you actually used, learning rate, batch size, warmup, adam's beta)?

BTW: "_Up scale the residual x with beta;_" the **beta** here should be **alpha**.

**Comment by maydaygmail** [comment]
Thanks @shumingma 
I will check the hyper parameters. Does the 1000 layers DeepNet model training need model parallel?

**Comment by donglixp** [comment]
@KennyShang @maydaygmail https://github.com/microsoft/torchscale
